### PR TITLE
zcash_address: Include HRP inside Unified Address Padding bytes

### PR DESCRIPTION
**Pull request by str4d**
Closes zcash/librustzcash#418.

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK

File: components/zcash_address/src/kind/unified.rs [hunk 1]
```diff
@@ -371,20 +382,20 @@ mod tests {
             0x7b, 0x28, 0x69, 0xc9, 0x84,
         ];
         assert_eq!(
-            Address::try_from(&invalid_padding[..]),
+            Address::try_from((MAINNET, &invalid_padding[..])),
             Err(ParseError::InvalidEncoding)
         );
 
         // Truncated padding ([0x00; 15] instead of [0x00; 16])
```
Comment (daira) [inline components/zcash_address/src/kind/unified.rs:389]:
```suggestion
        // Truncated padding ([0x00; 15])
```

**Review by daira — APPROVED** [review]
utACK with minor suggestion.

File: components/zcash_address/src/kind/unified.rs (hunk 1 above)
Comment (str4d) [inline components/zcash_address/src/kind/unified.rs:389]:
```suggestion
        // Short padding (padded to 15 bytes instead of 16)
```

**Review by daira — COMMENTED** [review]
Post-hoc ACK c02931dc821b8c08ac50ea5d944521dcbdb06035

**Pull request merged**